codegen/masm/src/emulator/memory.rs

```diff
@@ -0,0 +1,59 @@
+use std::{
+    fmt::Debug,
+    ops::{Index, IndexMut},
+};
+
+use miden_core::FieldElement;
+use midenc_hir::Felt;
+use rustc_hash::FxHashSet;
+
+const EMPTY_WORD: [Felt; 4] = [Felt::ZERO; 4];
+
+pub struct Memory {
+    memory: Vec<[Felt; 4]>,
+    set_memory_addrs: FxHashSet<usize>,
```

Comment:
*Suggestion:* I would make this a `BTreeSet`, so that you can print the memory addresses in order, plus its probably faster than a hashset in this case, since memory addresses are totally ordered, and highly likely to be written to in order from least to greatest

Reply:
Good catch! I missed it since I only dealt with a couple of addresses, but the order is important.
Done in https://github.com/0xPolygonMiden/compiler/pull/237/commits/8ca444b1a4e62a4faf11005216554dc58cac6a1a